pkg/rate/api_limiter.go

```diff
@@ -436,7 +438,7 @@ func (l *APILimiter) adjustedParallelRequests() int {
 	return int(l.adjustmentLimit(newParallelRequests, float64(l.params.ParallelRequests)))
 }
 
-func (l *APILimiter) requestFinished(r *limitedRequest, err error) {
+func (l *APILimiter) requestFinished(r *limitedRequest, err error, code string) {
```

Comment:
I feel like code should be provided as an integer and converted to a string before storing as a label, any reason your providing a string directly in the API?

Reply:
I tried adding the code as integer. Couple of things that made me change it to a string(atleast for the initial review):
1. Usage in `kvstore/etcd.go`, if I used `int` I wasn't sure what value should be passed from there. Easier to pass empty string then an arbitrary number.
2. As you mentioned, this is intended to be used in metrics and the metric function has `labelvalues` which accepts only string parameter. 
I am happy to change it `int` as it makes more sense to me as well, but want to know what should I return from `kvstore/etcd.go` in that case. Should that be a `0` ?

Reply:
0 seems reasonable, but I'm not familiar with that code path to know what makes the most sense. Alternative `-1` could be used if we're worried about `0` being a useful value in other cases. I don't think we want a pointer though.

Reply:
Thanks! Will wait for more feedback before incorporating this change, if that's okay ?